### PR TITLE
Use `StringNumber` for integers beyond `i32` - part2

### PR DESCRIPTION
**Pull request by marioCluml**
Closes : #261 ;

### CODE REVIEW

**Comment by marioCluml** [comment]
@sophie-cluml Looking at #257, it seems that for some functions with `u32`, it's wrapped in `ID` instead of `StringNumber`. I was wondering if those were two in the same thing or when is it applicable?

Also, I know that `usize`/`isize` can be bigger than 32-bits (64-bits depending on the target platform) however, how would I determine if they would need to be changed to `i32` or `StringNumber` (e.g. something like `error_code` ,`rule_id`, or `cluster_id`) or if the other int types (`u32`, etc.)  will fit within `i32`?

Would I need to make tests for that with `std::u32::MAX` and inserting and querying to check the fields? If yes, would I need to make a separate PR in `review-database` to change the fields that are say `u32` to `i32`?

**Comment by henry0715-dev** [comment]
@marioCluml 
The change was made because the tag_ids value is appropriate to express it as ID type rather than StringNumber(u32).

There is no need to change the value of i32.
However, in the case of u32, you need to change it to StringNumber because the positive integer value that can be displayed is larger than i32.

In the case of usize and isize as a type of GraphQL API response, it seems that conversion is also necessary.
However, for xxxTotalCount used for paging, you do not need to modify it.

**Comment by henry0715-dev** [comment]
@marioCluml 
The codes below also seem to need to be changed.
```
    async fn error_code(&self) -> u32 {
        self.inner.error_code
    }
    async fn request_len(&self) -> usize {
        self.inner.request_len
    }
    async fn response_len(&self) -> usize {
        self.inner.response_len
    }
```

**Comment by marioCluml** [comment]
> @marioCluml The codes below also seem to need to be changed.
> 
> ```
>     async fn error_code(&self) -> u32 {
>         self.inner.error_code
>     }
>     async fn request_len(&self) -> usize {
>         self.inner.request_len
>     }
>     async fn response_len(&self) -> usize {
>         self.inner.response_len
>     }
> ```

@henry0715-dev Done, I believe I fixed all instances in those files just was waiting for confirmation if I should use `ID` or `StringNumber` for something like `rule_id`.

**Comment by marioCluml** [comment]
@sophie-cluml @henry0715-dev Should I include part 3 and part 4 in this PR? It seems that since they are nearly the same it will be feasible and will need less PRs to review but if they do need to be separated I can do two separate PRs for part 3 and part 4 instead of including it in this one.

**Comment by sophie-cluml** [comment]
@marioCluml I think it would be nice to limit the target of this PR to a single issue, since I think this is your first PR for the similar issues. And personally, I prefer to make a PR for each issue if possible.

**Comment by henry0715-dev** [comment]
@marioCluml 
I think it would be better to add a description to each function.
Examples are as follows.
https://github.com/aicers/review-web/pull/257/files#diff-021a3134ce1a3114350da75bb628e336bd6df5835d3963b72b64bf4645869372R100

File: src/graphql/event/http.rs
Comment (henry0715-dev) [inline src/graphql/event/http.rs:117]: it is recommended to modify the document to end with a "."

Comment (marioCluml) [reply src/graphql/event/http.rs]: Done

**Review by henry0715-dev — APPROVED** [review]
LGTM

File: CHANGELOG.md
Comment (sehkone) [inline CHANGELOG.md:62]: To maintain consistency with `HttpThreat`, it would be better to use capital characters like `BlockListLdap` instead of `blockListLdap`. What do you think?

File: src/graphql/event/http.rs
Comment (sehkone) [inline src/graphql/event/http.rs:191]: I think `ID` is a better option.

File: src/graphql/event/ldap.rs
Comment (sehkone) [inline src/graphql/event/ldap.rs:178]: I suggest `ID`.

File: src/graphql/event/ldap.rs
Comment (sehkone) [inline src/graphql/event/ldap.rs:300]: `ID`

File: src/graphql/event/log.rs
Comment (sehkone) [inline src/graphql/event/log.rs:36]: `ID`

File: src/graphql/event/log.rs
Comment (sehkone) [inline src/graphql/event/log.rs:46]: `ID`

File: src/graphql/event/kerberos.rs
Comment (sehkone) [inline src/graphql/event/kerberos.rs:82]: @marioCluml @sophie-cluml,

I think we need to check if 'last_time' actually means what you described based on the protocol. If understanding the protocol takes time, it’s fine to omit this comment for now. The same approach should be applied to other protocol-specific terms.

File: src/graphql/event/http.rs
Comment (henry0715-dev) [reply src/graphql/event/http.rs:191]: @sehkone 
I have a question. In the case of TIDB, since the rule ID of TidbRule is obtained, I thought it would be correct to change it to ID. However, I couldn't find the source for the `rule_id` value in HTTP. It appears that this value is simply a `u32`, so I'm wondering if it is also correct to change this value to ID. 
- Specifically, I’m asking whether it should be changed to ID because it is expressed as the ID of `rule_id`.

File: src/graphql/event/http.rs
Comment (henry0715-dev) [inline src/graphql/event/http.rs:117]: If there is a new document you are adding, please remove it.
I also removed it while receiving a recent pre-review.

Comment (sehkone) [reply src/graphql/event/http.rs:117]: I asked Parker and Ben to come up with descriptive terms for every field in the relevant code, and we will use them later.

File: CHANGELOG.md
Comment (marioCluml) [reply CHANGELOG.md:62]: Sounds good. Will do 👍

File: src/graphql/event/http.rs
Comment (marioCluml) [reply src/graphql/event/http.rs:117]: For now, I'll remove the descriptions across parts 2, 3, and 4.

Comment (sehkone) [reply src/graphql/event/http.rs:117]: I think it would be more efficient to review the terms after Parker and Ben suggest them. This way, you don’t need to add any new ones for now. However, the terms you’ve already added can remain as they are.

Comment (marioCluml) [reply src/graphql/event/http.rs:117]: I've left the terms for now until the new terms are given. I can then add/change the descriptions once we get the new ones.

**Comment by sophie-cluml** [comment]
Could you rebase please?

**Comment by sophie-cluml** [comment]
I think https://github.com/aicers/review-web/pull/267/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR56 line refers to the similar task carried out by Henry, so I would like to suggest combining your CHANGELOG with this line.

**Comment by marioCluml** [comment]
> I think https://github.com/aicers/review-web/pull/267/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR56 line refers to the similar task carried out by Henry, so I would like to suggest combining your CHANGELOG with this line.

Done, Will apply to part 3 and part 4 sequentially after merges.

**Review by sophie-cluml — APPROVED** [review]
LGTM

**Pull request merged**
